substrait_consumer/consumers.py

```diff
@@ -146,3 +148,82 @@ def run_substrait_query(self, substrait_query: str) -> pa.Table:
         result = reader.read_all()
 
         return result
+
+
+class DatafusionConsumer:
+    """
+    Adapts the Datafusion Substrait consumer to the test framework.
+    """
+
+    def __init__(self):
+        self.ctx = SessionContext()
+
+    def setup(self, db_connection, created_tables, file_names: Iterable[str]):
+        if len(file_names) > 0:
+            parquet_file_paths = SubstraitUtils.get_full_path(file_names)
+            for file_name, file_path in zip(file_names, parquet_file_paths):
+                table_name = Path(file_name).stem
+                table_name = table_name.translate(
+                    str.maketrans("", "", string.punctuation)
+                )
+                if f"{self.__class__.__name__}{table_name}" not in created_tables:
+                    created_tables.add(f"{self.__class__.__name__}{table_name}")
+                    self.ctx.register_parquet(f"{table_name}", file_path)
+        else:
+            tables = pa.RecordBatch.from_arrays(
+                [
+                    pa.array([1, 2, 3, -4, 5, -6, 7, 8, 9, None]),
```

Comment:
If these are the same as other consumers perhaps we could extract this out?

Reply:
Good point.  Done!